components/collections/src/codepointinvlist/cpinvlist.rs

```diff
@@ -393,6 +393,31 @@ impl<'data> CodePointInversionList<'data> {
         })
     }
 
+    /// Yields an [`Iterator`] returning the ranges of the code points that are
+    /// *not* included in the [`CodePointInversionList`]
+    ///
+    /// Ranges are returned as [`RangeInclusive`], which is inclusive of its
+    /// `end` bound value. An end-inclusive behavior matches the ICU4C/J
+    /// behavior of ranges, ex: `CodePointInversionList::contains(UChar32 start, UChar32 end)`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use icu_collections::codepointinvlist::CodePointInversionList;
+    /// let example_list = [0x41, 0x44, 0x45, 0x46];
+    /// let example =
+    ///     CodePointInversionList::try_from_inversion_list_slice(&example_list)
+    ///         .unwrap();
+    /// let mut example_iter_ranges = example.iter_ranges_complemented();
+    /// assert_eq!(Some(0..=0x40), example_iter_ranges.next());
+    /// assert_eq!(Some(0x44..=0x44), example_iter_ranges.next());
+    /// assert_eq!(Some(0x46..=char::MAX as u32), example_iter_ranges.next());
+    /// assert_eq!(None, example_iter_ranges.next());
+    /// ```
+    pub fn iter_ranges_complemented(&self) -> impl Iterator<Item = RangeInclusive<u32>> + '_ {
```

Comment:
Thought: Another way to do this would be to return an iterator of `CodePointMapRange` with values of `true` or `false`, covering all code points

Reply:
This is one reason I was never super happy with the name `CodePointMapRange`; it should be `CodePointRangeWithValue` so that it can more uncontroversial be used in other contexts

Reply:
Yeah the thing is that becomes an API that is useful for exactly one thing: `iter_ranges_complemented()` (since `iter_ranges()` exists already)

I _guess_ it _might_ be useful for people wishing to construct CPTs out of invlists but I'd rather wait and see if someone asks for that.

Reply:
I was thinking about the convert-to-map use case. But it's fair to wait for that